redwood-lazylayout-widget/src/commonTest/kotlin/app/cash/redwood/lazylayout/widget/WindowedLazyListTest.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.lazylayout.widget
+
+import app.cash.redwood.Modifier
+import app.cash.redwood.layout.api.Constraint
+import app.cash.redwood.layout.api.CrossAxisAlignment
+import app.cash.redwood.lazylayout.api.ScrollItemIndex
+import app.cash.redwood.ui.Margin
+import app.cash.redwood.widget.MutableListChildren
+import app.cash.redwood.widget.Widget
+import app.cash.turbine.test
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import kotlin.test.Test
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+
+class WindowedLazyListTest {
+  @Test
+  fun updateViewPort() = runTest {
+    val windowedLazyList = FakeWindowedLazyList(this, "foo")
+    repeat(100) {
+      windowedLazyList.items.insert(
+        it,
+        object : Widget<String> {
+          override val value: String = it.toString()
+          override var modifier: Modifier = Modifier
+        },
+      )
+    }
+    windowedLazyList.updateViewport(29, 38)
+    windowedLazyList.viewportChanges.test {
+      assertThat(awaitItem()).isEqualTo(10..50)
+    }
+  }
+}
+
+private class FakeWindowedLazyList<W : Any>(
+  private val scope: CoroutineScope,
+  override val value: W,
+) : WindowedLazyList<W>(NoOpListUpdateCallback) {
+  private val _viewportChanges = MutableSharedFlow<IntRange>()
+  val viewportChanges: SharedFlow<IntRange> = _viewportChanges
+
+  override var modifier: Modifier = Modifier
+  override val placeholder: Widget.Children<W> = MutableListChildren()
+
+  init {
+    onViewportChanged { firstPagedItemIndex, lastPagedItemIndex ->
+      scope.launch {
+        _viewportChanges.emit(firstPagedItemIndex..lastPagedItemIndex)
+      }
+    }
+  }
+
+  override fun isVertical(isVertical: Boolean) = TODO()
```

Comment:
Will we ever do these todos? Maybe replace with `error("unexpected call")` ?

Reply:
Probably not! Replaced them